packages/executor/src/index.ts

```diff
@@ -453,10 +537,10 @@ export class ChugSplashExecutor extends BaseServiceV2<
 
         // If we make it to this point, we know that the executor has executed the bundle (or that it
         // has been cancelled by the owner), and that the executor has claimed its payment.
-
-        // Remove the current event from the events queue.
-        this.state.eventsQueue.pop()
       }
+
+      // Remove the current event from the events queue.
+      this.state.eventsQueue.pop()
```

Comment:
This is what was causing the infinite retry loop I noticed. If no active bundle is found in the project then the event would not be removed from the queue, and we'd get an infinite retry loop.